common/JailUtil.cpp

```diff
@@ -304,6 +304,63 @@ void setupChildRoot(bool bindMount, const std::string& childRoot, const std::str
                 "mount_jail_tree config in coolwsd.xml.");
 }
 
+/// Create a random device, either via mknod or by bind-mounting.
+bool createRandomDeviceInJail(const std::string& root, const std::string& devicePath, dev_t dev)
+{
+    const std::string absPath = root + devicePath;
+
+    if (FileUtil::Stat(absPath).exists())
+    {
+        LOG_DBG("Random device [" << devicePath << "] already exits");
+        return true;
+    }
+
+    LOG_DBG("Making [" << devicePath << "] node in [" << root << "/dev]");
+
+    if (mknod((absPath).c_str(),
+              S_IFCHR | S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH, dev) == 0)
+    {
+        LOG_DBG("Created random device [" << absPath << ']');
+        return true;
+    }
+
+    const auto mknodErrno = errno;
+
+    if (isBindMountingEnabled())
+    {
+        LOG_DBG("Failed to create random device via mknod("
+                << absPath << "). Mount must not use nodev flag. Will try bind-mounting instead: "
+                << strerror(mknodErrno));
+
+        Poco::File(absPath).createFile();
+        if (coolmount("-b", devicePath, absPath))
+        {
+            LOG_DBG("Bind mounted [" << devicePath << "] -> [" << absPath << ']');
```

Comment:
I would like a once-per-forkit or once-per-coolwsd LOG_WRN message in this case if we have to start doing this - lets have a static bool warned = false; if (!warned) LOG_WRN("Performance issue: nodev mount permission - have to bind mount random devices.") or something in there.